### PR TITLE
Fixing workspaces adding/editing

### LINKED ISSUE #1403: Can not create new workspace
workspace failure

### LINKED ISSUE #1404: Workspace entry tab doesn't save workspace name
**Describe the bug**

Workspace entry tab doesn't save workspace name

**To Reproduce**
1. Go to 'Workspace entry' (left-bottom corner)
2. Click on 'Workspace'
3. Click on 'New Workspace'
4. Click on 'Save'
5. Doesn't save as expected

**Expected behavior**

Save workspace name

**Screenshots**

Screen Recording 2020-11-17 at 11 32 11

**Environment (please complete the following information):**
- Lens Version: 4.0.0 beta.3
- OS: OSX Big Sur
- Installation method (e.g. snap or AppImage in Linux): Download .dmg
 
**Logs:**

from renderer process
```
mobx.module.js:1838 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@383]' TypeError: e.toJSON is not a function
    at workspace-store.ts:210
    at Array.map (<anonymous>)
    at r.value (workspace-store.ts:210)
    at base-store.ts:85
    at mobx.module.js:2163
    at ce (mobx.module.js:775)
    at e.track (mobx.module.js:1809)
    at g (mobx.module.js:2162)
    at e.onInvalidate (mobx.module.js:2150)
    at e.runReaction (mobx.module.js:1775)
e.reportExceptionInDerivation @ Lens.js:2
2mobx.module.js:1838 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@383]' TypeError: e.toJSON is not a function
    at workspace-store.ts:210
    at Array.map (<anonymous>)
    at r.value (workspace-store.ts:210)
    at base-store.ts:85
    at mobx.module.js:2163
    at ce (mobx.module.js:775)
    at e.track (mobx.module.js:1809)
    at g (mobx.module.js:2162)
    at e.onInvalidate (mobx.module.js:2150)
    at e.runReaction (mobx.module.js:1775)
```

**Additional context**

Error is from here https://github.com/lensapp/lens/blob/master/src/common/workspace-store.ts#L210

### PR DESCRIPTION
**Pull request by aleksfront**
* Fixing broken workspace addition
* Showing all workspaces in list but indicate disabled ones

adding workspace

Fixes #1404 
Fixes #1403

### CODE REVIEW

File: src/renderer/components/+workspaces/workspaces.tsx [hunk 1]
```diff
@@ -111,7 +115,7 @@ export class Workspaces extends React.Component {
           <Trans>Workspaces</Trans>
         </h2>
         <div className="items flex column gaps">
-          {this.workspaces.map(({ id: workspaceId, name, description, ownerRef }) => {
+          {this.workspaces.map(({ id: workspaceId, name, description, ownerRef, enabled }) => {
```
Comment (jakolehm) [inline src/renderer/components/+workspaces/workspaces.tsx]: `!enabled` should be hidden.

**Review by jakolehm — CHANGES_REQUESTED** [review]

File: src/renderer/components/+workspaces/workspaces.tsx (hunk 1 above)
Comment (aleksfront) [reply src/renderer/components/+workspaces/workspaces.tsx]: @jakolehm What is the reason for this? What we should do with significant delay before new workspace show up in the list?

Comment (jakolehm) [reply src/renderer/components/+workspaces/workspaces.tsx]: Store might have stale workspaces that extensions have created (`enabled === false`), we don't want to show them in UI.

Comment (aleksfront) [reply src/renderer/components/+workspaces/workspaces.tsx]: Fixed
add workspace

**Review by jakolehm — APPROVED** [review]

**Pull request merged**